plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,562 @@
+        mso_values = dict(
+            name=name,
+            description=description,
+            interfaceRoutingPolicyRef=interface_routing_policy_uuid,
+            qosPriority=qos_priority,
+            qosRef=custom_qos_policy_uuid,
+        )
+        if interface_routing_policy:
+            empty_interface_routing_policy = check_if_all_elements_are_none(list(interface_routing_policy.values()))
+            if not empty_interface_routing_policy:
+                mso_values["interfaceRoutingPolicyRef"] = mso_template.get_l3out_interface_routing_policy_uuid(
+                    mso, interface_routing_policy.get("name"), interface_routing_policy.get("template")
+                )
+        if custom_qos_policy:
+            empty_custom_qos_policy = check_if_all_elements_are_none(list(custom_qos_policy.values()))
+            if not empty_custom_qos_policy:
+                mso_values["qosRef"] = mso_template.get_custom_qos_policy_uuid(mso, custom_qos_policy.get("name"), custom_qos_policy.get("template"))
+
+        if match:
+            mso_values_remove = list()
+
+            if (interface_routing_policy_uuid == "" or empty_interface_routing_policy) and match.details.get("interfaceRoutingPolicyRef"):
+                mso_values_remove.append("interfaceRoutingPolicyRef")
+            if (custom_qos_policy_uuid == "" or empty_custom_qos_policy) and match.details.get("qosRef"):
+                mso_values_remove.append("qosRef")
+
+            for attribute_name, attribute_dict in attributes_dict.items():
+                if attribute_dict is not None:
+                    attribute_is_empty = check_if_all_elements_are_none(list(attribute_dict.values()))
+
+                    if match.details.get(attribute_name, {}).get("key", {}).get("ref"):
+                        match.details[attribute_name]["key"].pop("ref", None)
+                        mso.previous[attribute_name]["key"].pop("ref", None)
+                        mso.existing[attribute_name]["key"].pop("ref", None)
+
+                    if attribute_is_empty and match.details.get(attribute_name):
+                        mso_values_remove.append(attribute_name)
+
+                    elif not attribute_is_empty and not match.details.get(attribute_name):
+                        update_mso_values_attributes_payload(mso_values, {attribute_name: attribute_dict})
+
+                    elif not attribute_is_empty and match.details.get(attribute_name):
+                        mso_values[(attribute_name, "enabled")] = attribute_dict.get("enable")
+                        if attribute_dict.get("enable") is True:
+                            if attribute_name == "ospf":
+                                mso_values[(attribute_name, "authType")] = attribute_dict.get("authentication_type")
+                            else:
+                                mso_values[(attribute_name, "authEnabled")] = attribute_dict.get("authentication")
```

Comment:
in sabari his previous node group he uses ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP is that also needed here? might also apply to the other bool value

Reply:
It's not needed here because I directly use bool as input and pass it to the payload. I will change it if it becomes a consistency issue?